ciongke/cmd/hook/main_test.go

```diff
@@ -0,0 +1,308 @@
+/*
+Copyright 2016 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"net/http"
+	"net/http/httptest"
+	"strings"
+	"testing"
+
+	"github.com/kubernetes/test-infra/ciongke/github"
+
+	"k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/apis/batch"
+	"k8s.io/kubernetes/pkg/client/unversioned/testclient"
+	"k8s.io/kubernetes/pkg/runtime"
+)
+
+func TestServeHTTPErrors(t *testing.T) {
+	s := &Server{
+		Events:     make(chan Event),
+		HMACSecret: []byte("abc"),
+	}
+	// This is the SHA1 signature for payload "{}" and signature "abc"
```

Comment:
This might be easier to understand (now that I spent many minutes figuring it out)
`echo -n '{}' | openssl dgst -sha1 -hmac abc`